starlite/middleware/session.py

```diff
@@ -0,0 +1,250 @@
+    httponly: Optional[bool] = None
+    """Forbids javascript to access the cookie via 'Document.cookie'."""
+    samesite: Literal["lax", "strict", "none"] = "lax"
+    """Controls whether or not a cookie is sent with cross-site requests. Defaults to 'lax'."""
+
+    @validator("secret", always=True)
+    def validate_secret(cls, value: SecretBytes) -> SecretBytes:  # pylint: disable=no-self-argument
+        """Ensures that the 'secret' value is 128, 192 or 256 bits.
+
+        Args:
+            value: A bytes string.
+
+        Raises:
+            ValueError: if the bytes string is of incorrect length.
+
+        Returns:
+            A bytes string.
+        """
+        if len(value.get_secret_value()) not in [16, 24, 32]:
+            raise ValueError("secret length must be 16 (128 bit), 24 (192 bit) or 32 (256 bit)")
+        return value
+
+
+class SessionMiddleware(MiddlewareProtocol):
+    def __init__(
+        self,
+        app: "ASGIApp",
+        config: SessionCookieConfig,
+    ):
+        """Starlite SessionMiddleware.
+
+        Args:
+            app: The 'next' ASGI app to call.
+            config: SessionCookieConfig instance.
+        """
+        super().__init__(app)
+        self.app = app
+        self.config = config
+        self.aesgcm = AESGCM(config.secret.get_secret_value())
+
+    def dump_data(self, data: Any) -> List[bytes]:
+        """Given orjson serializable data, including pydantic models and numpy
+        types, dump it into a bytes string, encrypt, encode and split it into
+        chunks of the desirable size.
+
+        Args:
+            data: Data to serialize, encrypt, encode and chunk.
+
+        Notes:
+            - The returned list is composed of a chunks of a single base64 encoded
+                string that is encrypted using AES-CGM.
+
+        Returns:
+            List of encoded bytes string of a maximum length equal to the 'CHUNK_SIZE' constant.
+        """
+        serialized = dumps(data, default=Response.serializer, option=OPT_SERIALIZE_NUMPY)
+        associated_data = dumps({"timestamp": round(time.time()), "max_age": self.config.max_age})
+        nonce = urandom(NONCE_SIZE)
+        encrypted = self.aesgcm.encrypt(nonce, serialized, associated_data=associated_data)
+        encoded = b64encode(nonce + encrypted + b"associated_data=" + associated_data)
+        return [encoded[i : i + CHUNK_SIZE] for i in range(0, len(encoded), CHUNK_SIZE)]
+
+    @staticmethod
+    def _validate_session(associated_data: dict) -> bool:
+        """Verifies the validity of the session.
+
+        Args:
+            associated_data: A dictionary contains the timestamp of the AEAD Tag and max-age of the session.
+
+        Returns:
+            bool: True if the session is NOT expired else False.
+        """
+        timestamp = associated_data["timestamp"]
+        max_age = associated_data["max_age"]
+        if timestamp + max_age <= int(time.time()):
```

Comment:
You could simply store the expiration value before hand, and check it. 

Also, no need for an if / else clause here. 

`return timestamp + max_age <= int(time.time())` is a boolean

Reply:
Good point. This private method is unnecessary boilerplate.